server.js

```diff
@@ -23,58 +26,72 @@ app.use("/public", express.static(__dirname + "/public"));
+            }
+        }).reverse()
         if (!req.cookies.session) {
-            res.render('index');
+            res.render('forum', { posts: posts });
             return;
         }
-        //try to get the current session from DB. If there's not, clear the useless cookie.
         const currentSession = await Session.findById(req.cookies.session);
         if (!currentSession) {
             res.clearCookie('session');
-            res.render('index');
+            res.render('forum', { posts: posts });
             return;
         }
         //try to find the user based on the session. If not found, clear the useless cookie
         const currentUser = await User.findById(currentSession.user);
         if (!currentUser) {
             res.clearCookie('session');
-            res.render('index');
+            res.render('forum', { posts: posts });
             return
         }
-        // If it got here, it means that the user is authenticated and ready to go to personal profile page.
-        res.redirect('/profile');
+        res.render('forum', { user: currentUser, posts: posts });
     } catch (err) {
-        res.render('error', { "error message": err })
+        res.redirect('/')
     }
+    res.render("forum")
 })
 
-app.get("/profile", async (req, res) => {
+app.get("/profile/:id", async (req, res) => {
     try {
-        //check if there's a session. If not, direct to the loging main page. In case someone try to get to profile page without authentication.
-        if (!req.cookies.session) {
-            res.redirect('/')
-            return;
+        const pageUser = await User.findById(req.params.id);
+        if(!req.cookies.session) {
+            res.render('profile', { user: pageUser })
+            return
         }
-        // check if session is in the database. If not, direct to main page. (the / will clear the cookie)
         const currentSession = await Session.findById(req.cookies.session);
-        if (!currentSession) {
-            res.redirect('/')
+        const currentUser = await User.findById(currentSession.user);
+        if (req.params.id !== currentUser.id) {
+            if (!currentUser.admin) {
+                res.render(`profile`, { user: pageUser })
+                return;
+            }
+            res.render(`profile`, { user: pageUser, admin: true })
             return;
         }
-        // find user base on the session
-        const currentUser = await User.findById(currentSession.user);
-        if (!currentUser) {
-            res.redirect('/')
+        if (!currentUser.admin) {
+            res.render(`profile`, { user: currentUser, session: currentSession })
             return;
         }
-        // if it got here it means the user is authenticated.
-        res.render("profile", currentUser)
+        // if it got here it means the user is authenticated and admin.
+        res.render(`profile`, { user: currentUser, session: currentSession, admin: true })
```

Comment:
Correct me if I'm wrong, but I think the whole thing can be simplified to

```suggestion
        res.render(`profile`, { user: currentUser, session: currentSession, admin: currentUser.admin })
```

Reply:
sure, the admin thing can be fixed, but are you saying to remove all those lines? How will it know about currentUser and session?

Reply:
No, you still get the currentUser etc (or, you know, put it in a middleware), but you don't need all the `if` branches if you're going to render the same thing, just pass a variable.